File: packages/emitter/testcases/common/collections/list-initializer/ListInitializer.ts
import { int } from "@tsonic/core/types.js";
import { List } from "@tsonic/dotnet/System.Collections.Generic.js";

export class User {
  id: int;
  constructor(id: int) {
    this.id = id;
  }
}

export function makeInts(): List<int> {
  return new List<int>([1, 2, 3]);
}

export function makeStrings(): List<string> {
  return new List<string>(["a", "b"]);
}

export function makeUsers(): List<User> {
  const u1 = new User(1);
  const u2 = new User(2);
  return new List<User>([u1, u2]);
}

File: packages/emitter/testcases/common/expected/collections/list-initializer/ListInitializer.cs
namespace TestCases.common.collections.listinitializer
{
    public class User
    {
        public int id { get; set; }

        public User(int id)
            {
            this.id = id;
            }
    }

            public static class ListInitializer
            {
                public static global::System.Collections.Generic.List<int> makeInts()
                    {
                    return new global::System.Collections.Generic.List<int> { 1, 2, 3 };
                    }

                public static global::System.Collections.Generic.List<string> makeStrings()
                    {
                    return new global::System.Collections.Generic.List<string> { "a", "b" };
                    }

                public static global::System.Collections.Generic.List<User> makeUsers()
                    {
                    var u1 = new User(1);
                    var u2 = new User(2);
                    return new global::System.Collections.Generic.List<User> { u1, u2 };
                    }
            }
}